src/procedures/wildcam/wildcam.js

```diff
@@ -0,0 +1,183 @@
+Wildcam.getSpeciesList = function() {
+    return SPECIES;
+};
+
+/**
+ * Searches the database for wildlife camera entries.
+ * Each return value includes information about the contents of its associated image.
+ * You can pass an entry to :func:`Wildcam.getImage` to get the actual image being described.
+ * 
+ * @param {Date=} startDate The earliest date to include in the results. If omitted, no starting cutoff is used for filtering.
+ * @param {Date=} stopDate The latest date to include in the results. If omitted, no stopping cutoff is used for filtering.
+ * @param {Species=} species Filters results to only entries which contained the requested species. If omitted, no species filtering is performed.
+ * @param {Latitude=} latitude Filters results to only entries within a given distance from a central location. Requires ``longitude`` and ``radius`` also be set.
+ * @param {Longitude=} longitude Filters results to only entries within a given distance from a central location. Requires ``latitude`` and ``radius`` also be set.
+ * @param {BoundedNumber<0>=} radius Filters results to only entries within a given distance from a central location. Requires ``latitude`` and ``longitude`` also be set.
+ * @returns {Array<Object>} All data entries matching the search, in chronological order
+ */
+Wildcam.search = function (startDate = null, stopDate = null, species = null, latitude = null, longitude = null, radius = null) {
+    startDate = startDate ? +startDate : -Infinity;
+    stopDate = stopDate ? +stopDate : Infinity;
+    species = new Set(species ? [species] : SPECIES);
+
+    const center = { latitude, longitude };
+    if (latitude !== null || longitude !== null || radius !== null) {
+        if (latitude === null || longitude === null || radius === null) {
+            throw Error('Distance filtering requires three params: latitude, longitude, AND radius');
+        }
+    }
+
+    return DATA.filter(x => {
+        const time = +x.date;
+        if (time < startDate || time > stopDate) return false;
+        if (!x.species.some(y => species.has(y.name))) return false;
+        if (radius !== null) {
+            const dist = geolib.getDistance(center, { latitude: x.latitude, longitude: x.longitude });
+            if (dist > radius) return false;
+        }
+        return true;
+    });
+};
+
+/**
+ * Gets the image associated with a given entry.
+ * The provided entry should be exactly the format that was returned by one of the various search RPCs in this service.
+ * 
+ * @param {Object} entry The search entry to get an image of
+ * @returns {Image} The snapshot associated with the given entry
+ */
+Wildcam.getImage = async function(entry) {
+    const url = entry.imageUrl;
+    logger.info(`requesting image from: ${url}`);
+
+    let data = IMG_CACHE[url];
+    if (!data) {
+        logger.info('image not cached - downloading...');
+        data = IMG_CACHE[url] = (await axios({ url, method: 'GET', responseType: 'arraybuffer' })).data;
```

Comment:
Using the same caching library as the API consumer would be better. Ideally, caching would be able to be set as part of the RPC annotation and allow caching to happen client side, too.

Reply:
To expand on this a little, the existing caching library supports caching to the filesystem, too, rather than storing everything in memory. Furthermore, it has max size parameters and max age which would be nice to have. Making it an annotation could allow us to set the cache header on the client request, too.

Reply:
Ah, for some reason I always forget there's a lib for caching. Just switched over to that (with fs store) and changed to alpha.